lib/scsi_defs.h

```diff
@@ -0,0 +1,134 @@
+    kEnclosureServices = 0xd,
+    kSimplifiedDirectAccess = 0xe,
+    kOpticalCardReaderWriter = 0xf,
+    kBridgeControllerCommands = 0x10,
+    kObjectBasedStorage = 0x11,
+    kAutomationDriveInterface = 0x12,
+    kWellKnownLogicalUnit = 0x1e,
+    kUnknown = 0x1f,
+  };
+
+  // SCSI Reference Manual Table 62 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  // Field code of implemented version of the SPC standard 
+  enum class Version : uint8_t {
+    kNoStandard = 0x0,
+    kSpc = 0x3,
+    kSpc2 = 0x4,
+    kSpc3 = 0x5,
+    kSpc4 = 0x6,
+    kSpc5 = 0x7,
+  };
+
+  // SCSI Reference Manual https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  // Opt codes defined for SCSI commands supported by this project; this list may increase
+  enum class OpCode : uint8_t {
+    kReportLuns = 0xa0,
+    kRequestSense = 0x3,
+    kInquiry = 0x12,
+    kReadCapacity10 = 0x25,
+    kServiceActionIn = 0x9e,
+    kModeSense6 = 0x1a,
+    kMaintenanceIn = 0xa3,
+    kReadToc = 0x43,
+    kDoPreventAllowMediumRemoval = 0x1e,
+    kPersistentReserveIn = 0x5e,
+    kPersistentReserveOut = 0x5f,
+    kReserve6 = 0x16,
+    kRelease6 = 0x17,
+    kStartStopUnit = 0x1b,
+    kTestUnitReady = 0x0,
+  };
+
+  // SCSI Reference Manual Table 10 https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf
+  struct ControlByte {
+    uint8_t obsolete : 2;
+    bool naca : 1;
+    uint8_t reserved : 3;
+    uint8_t vendorSpecific : 2;
+  };
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1 Table 11
+  enum class SenseResponse: uint8_t {
+    kCurrentDescriptorError = 0x72,
+    kDeferredDescriptorError = 0x73,
+    kCurrentFixedError = 0x70,
+    kDeferredFixedError = 0x71
+  };
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 2.4.1.5 Table 28
+  enum class SenseKey : uint8_t {
+    kNoSense = 0x0,
+    kRecoveredError = 0x1,
+    kNotReady = 0x2,
+    kMediumError = 0x3,
+    kHardwareError = 0x4,
+    kIllegalRequest = 0x5,
+    kUnitAttention = 0x6,
+    kDataProtect = 0x7,
+    kBlankCheck = 0x8,
+    kVendorSpecific = 0x9,
+    kCopyAborted = 0xA,
+    kAbortedCommand = 0xB,
+    kReserved = 0xC,
+    kVolumeOverflow = 0xD,
+    kMiscompare = 0xE,
+    kCompleted = 0xF
+  };
+
+
+  // Refer to https://www.seagate.com/files/staticfiles/support/docs/manual/Interface%20manuals/100293068j.pdf , Section 3.6 Table 58
+  struct InquiryCommand {
+    OpCode opCode = OpCode::kInquiry;
+    bool evpd: 1; // Enable Vital Product Data (EVPD)
+    bool obsolete: 1; // formerly CMDDT
+    uint8_t reserved: 6;
+    uint8_t pageCode: 8;
+    uint16_t allocationLength: 16;
+    ControlByte controlByte;
+    // uint16_t padding: 16; // Don't need padding for memcpy
+
+  } ABSL_ATTRIBUTE_PACKED;
```

Comment:
Add `static_assert(sizeof(InquiryCommand) == ` with the expected size of the struct.

Reply:
good idea!